.github/workflows/release.yaml

```
@@ -262,91 +257,69 @@ jobs:
+          lipo -create -output "target/${{ env.PROFILE }}/${{ env.BINARY }}" "target/aarch64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }}" "target/x86_64-apple-darwin/${{ env.PROFILE }}/${{ env.BINARY }}"
 
       - name: Prepare package
         run: |
-          mkdir -p "${{ env.OUT_DIR }}"
-          cp "target/${{ env.TARGET }}/${{ env.PROFILE }}/${{ env.BINARY }}" "${{ env.OUT_DIR }}"
-          cp -r assets "${{ env.OUT_DIR }}" || true # Ignore error if `assets` folder does not exist
+          mkdir -p "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/assets"
+          cat > "build/${{ env.PACKAGE_NAME }}.app/Contents/Info.plist" << EOF
+            <?xml version="1.0" encoding="UTF-8"?>
+            <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+            <plist version="1.0">
+                <dict>
+                    <key>CFBundleDevelopmentRegion</key>
+                    <string>en</string>
+                    <key>CFBundleDisplayName</key>
+                    <string>${{ env.PACKAGE_NAME }}</string>
+                    <key>CFBundleExecutable</key>
+                    <string>${{ env.BINARY }}</string>
+                    <key>CFBundleIdentifier</key>
+                    <string>${{ env.OWNER }}.${{ env.PACKAGE_NAME}}</string>
+                    <key>CFBundleName</key>
+                    <string>${{ env.PACKAGE_NAME }}</string>
+                    <key>CFBundleShortVersionString</key>
+                    <string>${{ env.VERSION }}</string>
+                    <key>CFBundleVersion</key>
+                    <string>${{ env.VERSION }}</string>
+                    <key>CFBundleInfoDictionaryVersion</key>
+                    <string>6.0</string>
+                    <key>CFBundlePackageType</key>
+                    <string>APPL</string>
+                    <key>CFBundleSupportedPlatforms</key>
+                    <array>
+                        <string>MacOSX</string>
+                    </array>
+                </dict>
+            </plist>
+          EOF
+          cp -r assets/ "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/assets"|| true # Ignore error if `assets` folder does not exist
+          cp "target/${{ env.PROFILE }}/${{ env.BINARY }}" "build/${{ env.PACKAGE_NAME }}.app/Contents/MacOS/"
+          ln -s /Applications build/
+          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder build "build/${{ env.PACKAGE }}.dmg"
 
-      - name: Compress package
-        working-directory: ./build/${{ env.PACKAGE }}
-        run: |
-          hdiutil create -fs HFS+ -volname "${{ env.PACKAGE_NAME }}" -srcfolder "${{ env.PACKAGE_NAME }}.app" "../${{ env.PACKAGE }}.dmg"
 
       - name: Upload package to artifacts
         uses: actions/upload-artifact@v4
         with:
-          path: build/${{ env.PACKAGE }}.dmg
+          path: "build/${{ env.PACKAGE }}.dmg"
```

```suggestion
          path: build/${{ env.PACKAGE }}.dmg
```
I think quotes don't work here.